src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java

```diff
@@ -42,4 +50,31 @@ public void taskCompletedWithProblems(Executor executor, Queue.Task task, long d
     public String toString() {
         return String.format("KubernetesComputer name: %s slave: %s", getName(), getNode());
     }
-}
\ No newline at end of file
+
+    private void annotatePodWithTaskInfo(Queue.Task task, String status) {
+        KubernetesClient k8sClient = null;
+        try {
+            KubernetesSlave slave = getNode();
+            if (slave != null) {
+                k8sClient = slave.getKubernetesCloud().connect();
+                Pod pod = k8sClient.pods().withName(slave.getNodeName()).edit()
+                        .editMetadata()
+                        .addToAnnotations(TASK_NAME_POD_ANNOTATION, task.getName())
+                        .addToAnnotations(TASK_STATUS_POD_ANNOTATION, status)
+                        .endMetadata().done();
+
+                String nodeName = pod.getSpec().getNodeName();
+                LOGGER.info("task [" + task.getName() + "] updated with status [" + status + "] in pod [" + slave.getNodeName() + "] on node ["
```

Comment:
It's preferred for performance to use the method with arguments

    LOGGER.log(Level.FINE, "task {0} updated with status {1} ...", new String [] {task.getName(), status,...} )

I would use FINE